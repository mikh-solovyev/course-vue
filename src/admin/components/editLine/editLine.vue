<template>
  <div class="edit-line-component" :class="{'blocked' : blocked}">
    <div class="title" v-if="editmode === false">
      <div class="text">{{value}}</div>
      <div class="icon">
        <icon symbol="pencil" grayscale @click="editmode = true"></icon>
      </div>
    </div>
    <div v-else class="title">
      <div class="input">
        <app-input
          placeholder="Название новой группы"
          :value="value"
          :errorText="errorText"
          :errorMessage="errorMessage"
          @input="$emit('input', $event)"
          @keydown.native.enter="onApprove"
          autofocus="autofocus"
          no-side-paddings="no-side-paddings"
        ></app-input>
      </div>
      <div class="buttons">
        <div class="button-icon">
          <icon symbol="tick" @click="onApprove"></icon>
        </div>
        <div class="button-icon">
          <icon symbol="cross" @click="$emit('remove')"></icon>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import appInput from "../input";
import icon from "../icon";

export default {
  props: {
    value: {
      type: String,
      default: ""
    },
    errorText: {
      type: String,
      default: ""
    },
    editModelByDefault: Boolean,
    blocked: Boolean
  },
  data() {
    return {
      editmode: this.editModelByDefault,
      title: this.value,
      errorMessage: ""
    };
  },
  methods: {
    onApprove() {
      if(this.value.trim() === "") {
        this.errorMessage = "Введите название"
        return false;
      }

      if (this.title.trim() === "") return false;

      if (this.title.trim() === this.value.trim()) {
        this.editmode = false;
      } else {
        this.$emit("approve", this.value);
        this.errorMessage = "";
      }
    }
  },
  components: {
    icon,
    appInput
  }
};
</script>

<style lang="postcss" scoped src="./editLine.pcss"></style>
