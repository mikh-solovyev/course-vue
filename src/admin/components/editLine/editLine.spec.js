import editLine from "./editLine.vue";
import icon from "../icon/icon.vue";
import { mount } from "@vue/test-utils";

it("component render", async () => {
    const wrapper = mount(editLine);
    const component = await wrapper.find(".edit-line-component");

    expect(component.exists()).toBe(true);
});

it("editmode worked", async () => {
    const wrapper = mount(editLine);

    await wrapper.setProps({
        editmode: false
    });

    const icons = await wrapper.findComponent(icon);
    await icons.trigger("click");

    const editedInput = await wrapper.find(".input");
    expect(editedInput.exists()).toBe(true);
});

it("snapshots", async () => {
    const wrapper = mount(editLine);

    expect(wrapper.element).toMatchSnapshot();
});