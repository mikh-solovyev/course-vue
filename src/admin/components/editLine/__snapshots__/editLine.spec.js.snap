// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<div
  class="edit-line-component"
>
  <div
    class="title"
  >
    <div
      class="text"
    >
      
    </div>
     
    <div
      class="icon"
    >
      <button
        class="icon-component is-pencil grayscale no-words"
        data-text=""
        type="button"
      />
    </div>
  </div>
</div>
`;
